### DISCUSSION TITLE
How to implement and use a modal Dialog in a custom extension?

### DISCUSSION THREAD

**Discussion by Pachat** [directus/directus#13695] [Archive]
Do I have to do it directly as a vue dialog
or, preferably, is there a way to base it on Directus modal dialog?
In that case, what to import in the extension?

**Comment by br41nslug** [comment]
Directus injects the components mentioned in the wiki so you should just be able to use `v-dialog` https://github.com/directus/directus/wiki/v-dialog

**Comment by Pachat** [comment]
Thank you for the hint.
Where do I put 
```
<v-dialog>
	<v-sheet>
		<h2>Are you sure you want to delete this item?</h2>
		<v-button>No</v-button>
		<v-button>Yes</v-button>
	</v-sheet>
</v-dialog>

```
in my component?
I tried in `module.vue`, but see nothing.

What I'd like is to open the dialog on the click on a button, and get the Yes/No answer to proceed.

**Reply by rijkvanzanten** [reply]
You don't see anything, as the dialog is disabled by default 🙂 Try:

```html
<v-dialog>
	<template #activator="{ on }">
		<v-button @click="on">Open Dialog</v-button>
	</template>

	<v-card>
		<v-card-title>Are you sure you want to delete this item?</v-card-title>
		<v-card-actions>
			<v-button secondary>No</v-button>
			<v-button>Yes</v-button>
		</v-card-actions>
	</v-card>
</v-dialog>
```
